R/subscript-tables.R

```diff
@@ -147,7 +155,29 @@ updateTableAttributes <- function(y, x, called.args) {
     attr.names <- names(attributes(y))
     names.needing.update <- !isBasicAttribute(attr.names)
     names(attributes(y))[names.needing.update] <- paste0("original.", attr.names[names.needing.update])
+    y <- updateSpanIfNecessary(y, x.attributes, called.args, evaluated.args)
     attr(y, "name") <- paste0(x.attributes[["name"]], "[",
                               paste(as.character(called.args), collapse = ","), "]")
     y
 }
+
+subscriptSpanDF <- function(span.attr, idx) {
+    if (is.character(idx))
+        idx <- which(span.attr[[2]] %in% idx)
+    out <- span.attr[idx, , drop = FALSE]
+    if (all(is.na(out[[1L]]))) invisible() else out
+}
+
+updateSpanIfNecessary <- function(y, x.attributes, called.args, evaluated.args) {
+    span.attribute <- x.attributes[["span"]]
+    if (is.null(span.attribute)) return(y)
+    dim.length <- length(x.attributes[["dim"]])
+    if (dim.length > 1L && length(evaluated.args) == 1L) return(y)
```

Comment:
Will add comment about single integer vector will drop spans if array has more than one dim.

Reply:
Add support for integer matrix (with d columns (1 or 2d)) and boolean matrix (only if bool mat is same size)

Reply:
Dropping support for integer matrices and boolean matrices. They don't retain their dimension in base R with that approach. 
E.g. the subscriptted array below will be a vector and not a matrix with col names and row names
```
x <- array(1:12, dim = 3:4, dimnames = list(LETTERS[1:3], letters[1:4]))
ind <- array(FALSE, dim = 3:4)
ind[1:2, ] <- TRUE
x[ind]
x[1:2, ]
```